src/compiler/checker.ts

```diff
@@ -6227,6 +6252,26 @@ module ts {
             return node === conditional.whenTrue || node === conditional.whenFalse ? getContextualType(conditional) : undefined;
         }
 
+        function getContextualTypeForJsxExpression(expr: JsxExpression|JsxSpreadAttribute): Type {
+            // Contextual type only applies to JSX expressions that are in attribute assignments (not in 'Children' positions)
+            if (expr.parent.kind === SyntaxKind.JsxAttribute) {
+                let attrib = <JsxAttribute>expr.parent;
+                let attrsType = getJsxElementAttributesType(<JsxOpeningElement>attrib.parent);

```

Comment:
This took me a while to understand. I thought you were getting the global ElementAttributesType. Instead, you are getting the type of the property denoted by the single property name in ElementAttributesType, off of the parent's class. Any way you could adjust the naming to reflect this?